holoviews/core/data/grid.py

```diff
@@ -503,7 +514,7 @@ def sample(cls, dataset, samples=[]):
             for vdim, array in zip(dataset.vdims, arrays):
                 flat_index = np.ravel_multi_index(tuple(int_inds)[::-1], array.shape)
                 if da and isinstance(array, da.Array):
-                    data[vdim.name].append(array.flatten()[tuple(flat_index)])
+                    data[vdim.name].append(array.vindex[tuple(flat_index)])
```

Comment:
FWIW `flatten` is lazy and returns a Dask Array. So you could do both if needed.

Reply:
>FWIW flatten is lazy and returns a Dask Array. So you could do both if needed.

Thanks, I think that might be needed here. I'm wondering why the tests didn't error though, going to have to make sure the ``sample`` tests are actually working.